requirements-base.txt

```diff
@@ -73,7 +75,8 @@ grpcio==1.35.0
 hiredis==0.3.1
 
 # not directly used, but pinned for at least semaphore/symbolic
-cffi==1.12.2
+cffi==1.12.2; python_version == '3.6'
+cffi==1.14.4; python_version > '3.6'
```

Comment:
This is to help with Python 3.9 (affecting at least Linux):
https://github.com/getsentry/sentry/pull/24394/checks?check_run_id=2097113005
```
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/home/linuxbrew/.linuxbrew/Cellar/libffi/3.3_3/include -I/home/runner/work/sentry/sentry/.venv/include -I/opt/hostedtoolcache/Python/3.9.2/x64/include/python3.9 -c c/_cffi_backend.c -o build/temp.linux-x86_64-3.9/c/_cffi_backend.o
    c/_cffi_backend.c: In function ‘prepare_callback_info_tuple’:
    c/_cffi_backend.c:5987:5: warning: ‘PyEval_InitThreads’ is deprecated [-Wdeprecated-declarations]
     5987 |     PyEval_InitThreads();
          |     ^~~~~~~~~~~~~~~~~~
    In file included from /opt/hostedtoolcache/Python/3.9.2/x64/include/python3.9/Python.h:145,
                     from c/_cffi_backend.c:2:
    /opt/hostedtoolcache/Python/3.9.2/x64/include/python3.9/ceval.h:130:37: note: declared here
      130 | Py_DEPRECATED(3.9) PyAPI_FUNC(void) PyEval_InitThreads(void);
          |                                     ^~~~~~~~~~~~~~~~~~
    c/_cffi_backend.c: In function ‘b_callback’:
    c/_cffi_backend.c:6042:5: warning: ‘ffi_prep_closure’ is deprecated: use ffi_prep_closure_loc instead [-Wdeprecated-declarations]
     6042 |     if (ffi_prep_closure(closure, &cif_descr->cif,
          |     ^~
    In file included from c/_cffi_backend.c:15:
    /home/linuxbrew/.linuxbrew/Cellar/libffi/3.3_3/include/ffi.h:334:1: note: declared here
      334 | ffi_prep_closure (ffi_closure*,
          | ^~~~~~~~~~~~~~~~
    In file included from c/cffi1_module.c:20,
                     from c/_cffi_backend.c:7552:
    c/call_python.c: In function ‘_get_interpstate_dict’:
    c/call_python.c:26:30: error: dereferencing pointer to incomplete type ‘PyInterpreterState’ {aka ‘struct _is’}
       26 |     builtins = tstate->interp->builtins;
          |                              ^~
    error: command '/usr/bin/gcc' failed with exit code 1
```